templates/commands/render.go

```diff
@@ -454,11 +454,38 @@ func (c *RenderCommand) checkInputsMissing(spec *model.Spec) []string {
 func executeSpec(ctx context.Context, spec *model.Spec, sp *stepParams) error {
 	logger := logging.FromContext(ctx).Named("executeSpec")
 
-	for _, step := range spec.Steps {
+	for i, step := range spec.Steps {
 		if err := executeOneStep(ctx, step, sp); err != nil {
 			return err
 		}
 		logger.Debugw("completed template action", "action", step.Action.Val)
+		if sp.flags.DebugScratchContents {
+			sb := &strings.Builder{}
```

Comment:
Nit: it might be nice to move this to a function that returns a `string` so this becomes:

```go
if sp.flags.DebugScratchContents {
  logger.Info(debugScratch(sp.scratchDir))
}
```

Reply:
Done.